dcm4che-core/src/main/java/org/dcm4che3/io/stream/DelegateImageInputStream.java

```diff
@@ -0,0 +1,264 @@
+    public String readUTF() throws IOException {
+        return iis.readUTF();
+    }
+
+    @Override
+    public void readFully(byte[] b, int off, int len) throws IOException {
+        iis.readFully(b,off,len);
+    }
+
+    @Override
+    public void readFully(byte[] b) throws IOException {
+        iis.readFully(b);
+    }
+
+    @Override
+    public void readFully(short[] s, int off, int len) throws IOException {
+        iis.readFully(s,off,len);
+    }
+
+    @Override
+    public void readFully(char[] c, int off, int len) throws IOException {
+        iis.readFully(c,off,len);
+    }
+
+    @Override
+    public void readFully(int[] i, int off, int len) throws IOException {
+        iis.readFully(i,off,len);
+    }
+
+    @Override
+    public void readFully(long[] l, int off, int len) throws IOException {
+        iis.readFully(l,off,len);
+    }
+
+    @Override
+    public void readFully(float[] f, int off, int len) throws IOException {
+        iis.readFully(f,off,len);
+    }
+
+    @Override
+    public void readFully(double[] d, int off, int len) throws IOException {
+        iis.readFully(d,off,len);
+    }
+
+    @Override
+    public long getStreamPosition() throws IOException {
+        return this.iis.getStreamPosition();
+    }
+
+    @Override
+    public int getBitOffset() throws IOException {
+        return iis.getBitOffset();
+    }
+
+    @Override
+    public void setBitOffset(int bitOffset) throws IOException {
+        iis.setBitOffset(bitOffset);
+    }
+
+    @Override
+    public int readBit() throws IOException {
+        return iis.readBit();
+    }
+
+    @Override
+    public long readBits(int numBits) throws IOException {
+        return iis.readBits(numBits);
+    }
+
+    @Override
+    public long length() throws IOException {
+        return iis.length();
+    }
+
+    @Override
+    public int skipBytes(int n) throws IOException {
+        return iis.skipBytes(n);
+    }
+
+    @Override
+    public long skipBytes(long n) throws IOException {
+        return iis.skipBytes(n);
+    }
+
+    @Override
+    public void seek(long pos) throws IOException {
+        iis.seek(pos);
+    }
+
+    @Override
+    public void mark() {
+        iis.mark();
+    }
+
+    @Override
+    public void reset() throws IOException {
+        iis.reset();
+    }
+
+    @Override
+    public void flushBefore(long pos) throws IOException {
+        iis.flushBefore(pos);
+    }
+
+    @Override
+    public void flush() throws IOException {
+        iis.flush();
+    }
+
+    @Override
+    public long getFlushedPosition() {
+        return iis.getFlushedPosition();
+    }
+
+    @Override
+    public boolean isCached() {
+        return iis.isCached();
+    }
+
+    @Override
+    public boolean isCachedMemory() {
+        return iis.isCachedMemory();
+    }
+
+    @Override
+    public boolean isCachedFile() {
+        return iis.isCachedFile();
+    }
+
+    @Override
+    public void close() throws IOException {
```

Comment:
Should you set the iis to null here so that it is immediately known that this is closed if you try to re-use it?  I'm just concerend about someone using one of these and not realizing that it wasn't being used correctly.

Reply:
mmmm, it's worth being safe than sorry here.  If I add an accessor for iis then I can do that with minimal impact to the class.  I'll do that in my local branch.

Reply:
I updated the code to throw an IllegalStateException if the delegate is used after the close() is invoked.  That's not the best exception to throw, but i needed something that was Runtime so it could be thrown by all methods, and I didn't want to add a new exception for something this small...